plugins/enhanced_titles/__init__.py

```diff
@@ -0,0 +1,429 @@
+            name (str): Title of the album/track.
+
+        Returns:
+            (str): The sort name of the first useful alias it finds. None if
+                   none are found.
+
+        For example, "The Beatles" has alias "Le Double Blanc" with sort name
+        "Double Blanc, Le", so it's not considered. But it also has alias
+        "The Beatles" with sort name "Beatles, The", so this one is chosen.
+        Another example, "The Continuing Story of Bungalow Bill" has an alias
+        with the sort name equal to the title, this is not considered because
+        it makes more sense to swap the prefix.
+        """
+        for alias in aliases:
+            sortname = alias["sort-name"]
+            if (alias["name"].casefold() == name.casefold() and
+                    not sortname.casefold() == name.casefold()):
+                log.info("Enhanced Titles: sort name found for \"" + name + "\", \"" + sortname + "\".")
+                return sortname
+        log.info("Enhanced Titles: no proper sort name found for \"" + name + "\".")
+        return None
+
+    def _response_handler(self, document, reply, error, metadata = None, field = None):
+        """Handles the response from MusicBrainz.
+
+        Args:
+            metadata (MetaData): The object that needs to be updated.
+            field (str): Either "title" or "album", depending on what is being
+                         updated.
+        """
+        sortname = ""
+        try:
+            if document:
+                if error:
+                    log.error("Enhanced Titles: information retrieval error.")
+                if document["aliases"]:
+                    sortname = self._select_alias(document["aliases"], metadata[field])
+                else:
+                    log.info("Enhanced Titles: no aliases found for \"" + metadata[field] + "\".")
+        finally:
+            if sortname:
+                sortfield = field + "sort"
+                metadata[sortfield] = sortname
+            else:
+                self._swapprefix(metadata, field)
+
+    def _swapprefix(self, metadata, field):
+        """Swaps the prefix of the title based on the album/track language."
+
+        If no language information is found, then it uses all languages available.
+        Otherwise, if none of the languages are available, it just copies the title.
+        Otherwise it uses exclusively the languages that are also available.
+        """
+        sortfield = field + "sort"
+        languages = [metadata["language"], metadata["_releaselanguage"]]
+        languages = [language for language in languages if language]
+        if not languages:
+            prefixes = _articles[""] + [article.capitalize() for article in _articles[""]]
+            metadata[sortfield] = func_swapprefix(None, metadata[field], *prefixes)
+        else:
+            languages = [language for language in languages if language in _articles.keys()]
```

Comment:
Isn't this equivalent to:

```suggestion
            languages = set(languages).intersection(_articles)
```